pkg/streamingpromql/operators/aggregations/limitklimitratio/operator.go

```diff
@@ -388,11 +319,25 @@ type groupLimiterRatio struct {
 	seriesHashMap  map[int]float64
 }
 
-func (q *groupLimiterRatio) incSeriesRefCount() {
+func (q *groupLimiterRatio) increaseSeriesCount() {
 	q.seriesRefCount++
 }
 
-func (q *groupLimiterRatio) accumulateSeries() bool { return true }
+func (q *groupLimiterRatio) limitSeriesData(seriesIndex int, value types.InstantVectorSeriesData, timeRange types.QueryTimeRange) (types.InstantVectorSeriesData, error) {
+
+	value.Histograms = slices.DeleteFunc(value.Histograms, func(p promql.HPoint) bool {
+		step := (p.T - timeRange.StartT) / timeRange.IntervalMilliseconds
```

Comment:
```suggestion
		step := timeRange.PointIndex(p.T)
```

(same applies to histograms case below, as well as `groupLimiterK.limitSeriesData`)